src/Util/YamlSourceManipulator.php

```diff
@@ -623,7 +643,7 @@ private function getEndOfPreviousKeyPosition($key): int
 
     private function getKeyRegex($key)
     {
-        return sprintf('#%s( )*:#', preg_quote($key));
+        return sprintf('#(- )?%s( )*:#', preg_quote($key));
```

Comment:
We either have too many changes in this PR, or we're missing some more tests :).

If I revert this change AND revert the changes from lines 530-538, the new test PASSES. It seems that only the code from 559-568 is actually needed. If the other code fixes some other bug, we need a test case for that.

Reply:
You're right. I think this line was a "first resolution try" and I forgot it :)